sql/core/src/main/scala/org/apache/spark/sql/parquet/ParquetTypes.scala

```diff
@@ -48,8 +48,10 @@ private[parquet] case class ParquetTypeInfo(
   length: Option[Int] = None)
 
 private[parquet] object ParquetTypesConverter extends Logging {
-  def isPrimitiveType(ctype: DataType): Boolean =
-    classOf[PrimitiveType] isAssignableFrom ctype.getClass
+  def isPrimitiveType(ctype: DataType): Boolean = ctype match {
+    case _: NumericType | BooleanType | StringType | BinaryType => true
```

Comment:
We need to put all types other than `MapType`, `ArrayType`, and `StructType` here. Parquet uses this function to create record converters [here] [1].

[1]: https://github.com/apache/spark/blob/03e85b4a11899f37424cd6e1f8d71f1d704c90bb/sql/core/src/main/scala/org/apache/spark/sql/parquet/ParquetConverter.scala#L161-L165